tests/kafkatest/tests/streams/streams_application_upgrade_test.py

```diff
@@ -56,9 +55,9 @@ def perform_broker_upgrade(self, to_version):
             node.version = KafkaVersion(to_version)
             self.kafka.start_node(node)
 
-    @cluster(num_nodes=6)
-    @matrix(from_version=smoke_test_versions, bounce_type=["full"])
-    def test_app_upgrade(self, from_version, bounce_type):
+    @cluster(num_nodes=9)
```

Comment:
Why do we need 9 nodes now?

Reply:
I am still confuse why we need 3 more node now... no need to block the PR on this question, but I am just curious to understand why. -- I mean, if we have 3 brokers, and 3 controllers, plus 3 KS, yes, we would need 9. But before, we should have had 3 brokers, 3 ZK, and 3 KS, and thus also need 9? -- Are we actually running with dedicate quorum nodes? Or did this test not us ZK but "mixed-moded" broker/controller nodes before?

Reply:
Did we figure this one out?

Reply:
No I have not looked into this yet, probably ok to defer this to a follow-up task

Reply:
SG. -- We should just not drop it to ensure we run system test as efficient/light weight as possible.